include/deal.II/non_matching/quadrature_generator.h

```diff
@@ -456,6 +477,85 @@ namespace NonMatching
      * This quadrature always contains zero points in 1D.
      */
     const ImmersedSurfaceQuadrature<0, 1> surface_quadrature;
+
+    /**
+     * Function that describes our level set function in reference space.
+     */
+    std::unique_ptr<
+      internal::QuadratureGeneratorImplementation::CellWiseFunction<1>>
+      reference_space_level_set;
+  };
+
+
+  /**
+   * This class creates immersed quadrature rules of cells in reference space
+   * when the domain is described by a discrete level set vector.
+   */
+  template <int dim>
+  class DiscreteQuadratureGenerator : public QuadratureGenerator<dim>
+  {
+  public:
+    using AdditionalData = AdditionalQGeneratorData;
+
```

Comment:
Here too:
```suggestion
    /**
     * Constructor, the discrete level set functions is described by the
     * incoming DoFHandler and Vector. Pointers to the these are stored
     * internally, so they must have a longer lifetime than the created this
     * class. The hp::QCollection<1> and AdditionalData is passed to the
     * QuadratureGenerator class.
     */
```

Reply:
I think you should delete "the" in @simonsticko's suggestion "Pointers to the these "